packages/flutter/test/widgets/page_view_test.dart

```diff
@@ -908,4 +908,68 @@ void main() {
 
     semantics.dispose();
   });
+
+  testWidgets('Fire onScrollStarted when user starts to scroll ',
+      (WidgetTester tester) async {
+    int onScrollStartedCalledCount = 0;
+    await tester.pumpWidget(Directionality(
+      textDirection: TextDirection.ltr,
+      child: PageView(
+        onScrollStarted: () {
+          onScrollStartedCalledCount++;
+        },
+        children: kStates.map<Widget>((String state) => Text(state)).toList(),
+      ),
+    ));
+
+    final TestGesture gesture =
+        await tester.startGesture(const Offset(100.0, 100.0));
+    await gesture.moveBy(const Offset(-380.0, 0.0));
+    await tester.pump();
+
+    expect(onScrollStartedCalledCount, 1);
+  });
+
+  testWidgets('Fire onScrollCancelled when user cancels scroll ',
+      (WidgetTester tester) async {
+    int onScrollCancelledCalledCount = 0;
+    await tester.pumpWidget(Directionality(
+      textDirection: TextDirection.ltr,
+      child: PageView(
+        onScrollCancelled: () {
+          onScrollCancelledCalledCount++;
+        },
+        children: kStates.map<Widget>((String state) => Text(state)).toList(),
+      ),
+    ));
+
+    // Fling at a low speed and with short distance
+    await tester.fling(find.byType(PageView), const Offset(-200.0, 0.0), 10.0);
+    await tester.pumpAndSettle();
+
+    expect(onScrollCancelledCalledCount, 1);
+    expect(find.text('Alabama'), findsOneWidget);
+  });
+
```

Comment:
It's missing a test case where you do a drag, but the drag is not enough to change the page.